web3/eth.py

```diff
@@ -530,6 +530,15 @@ async def get_code(
     ) -> HexBytes:
         return await self._get_code(account, block_identifier)
 
+    get_block_transaction_count: Method[Callable[[BlockIdentifier], int]] = Method(
```

Comment:
This looks like a copy of the sync property with the same name... is there a reason we can't move this to `BaseEth` and keep one copy? 🤔

Reply:
I was just looking at that 👀 this PR triggered the thought I had about standardizing `AsyncEth` and `Eth`, which led to the typing conversation... We can just put this in `BaseEth` for now, but I think it will need to be moved/expanded on when we figure out what we want to do about our types.

Reply:
Yeah that makes sense 👍🏼. I'd be good with moving it to `BaseEth` for now while we explore the refactoring a bit more.